geomstats/riemannian_metric.py

```diff
@@ -50,12 +54,12 @@ def norm(self, vector, base_point=None):
         """
         Norm associated to the inner product.
         """
-        sq_norm = self.squared_norm(vector, base_point)
-        if sq_norm < 0:
+        n_negative_eigenvalues = self.signature[1]
```

Comment:
curious about your signature mechanism. Do you explain it somewhere in the code? Sorry if I lost track. Maybe it would be cool to encapsulate this in some kind of object so that it would be more clear for checks and internal behaviors?